File: OnlineShopAppEcommerce/app/src/main/java/com/pomenyatnazvaniedlyagoogleplay/onlineshopappecommerce/Activity/MainActivity.java
package com.pomenyatnazvaniedlyagoogleplay.onlineshopappecommerce.Activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import com.pomenyatnazvaniedlyagoogleplay.onlineshopappecommerce.Adapter.PopularListAdapter;
import com.pomenyatnazvaniedlyagoogleplay.onlineshopappecommerce.Domain.PopularDomain;
import com.pomenyatnazvaniedlyagoogleplay.onlineshopappecommerce.R;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView.Adapter adapterPopular;
    private RecyclerView recyclerViewPopular;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        initRecyclerView();
        bottom_navigation();

    }

    private void bottom_navigation() {
        LinearLayout homeBtn = findViewById(R.id.homeBtn);
        LinearLayout cartBtn = findViewById(R.id.cartBtn);


        //создание кнопки хз компас
        homeBtn.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, MainActivity.class)));
        //создание кнопки корзина
        cartBtn.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, CartActivity.class)));
    }

    private void initRecyclerView() {//список с данными
        ArrayList<PopularDomain> items = new ArrayList<>();
        items.add(new PopularDomain("Mac Book", "\"macOS is the most advanced\n" +
                "                desktop operating system in the world.\n" +
                "                 And with macOS Sonoma, work and play\n" +
                "                  on your Mac are even more powerful —\n" +
                "                  with new ways to elevate your video\n" +
                "                  presentations, boost your gaming\n" +
                "                  performance, and personalize your\n" +
                "                  device.\"", "pic1", 15, 4, 500));
        items.add(new PopularDomain("PS-5", "\"macOS is the most advanced\n" +
                "                desktop operating system in the world.\n" +
                "                 And with macOS Sonoma, work and play\n" +
                "                  on your Mac are even more powerful —\n" +
                "                  with new ways to elevate your video\n" +
                "                  presentations, boost your gaming\n" +
                "                  performance, and personalize your\n" +
                "                  device.\"", "pic2", 10, 4.5, 450));
        items.add(new PopularDomain("Phpne 14", "\"macOS is the most advanced\n" +
                "                desktop operating system in the world.\n" +
                "                 And with macOS Sonoma, work and play\n" +
                "                  on your Mac are even more powerful —\n" +
                "                  with new ways to elevate your video\n" +
                "                  presentations, boost your gaming\n" +
                "                  performance, and personalize your\n" +
                "                  device.\"", "pic3", 13, 4.2, 800));

        recyclerViewPopular = findViewById(R.id.view1);//ссылка на ресайкл вью
        recyclerViewPopular.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false));

        adapterPopular = new PopularListAdapter(items);
        recyclerViewPopular.setAdapter(adapterPopular);
    }

}

File: Notes/app/src/main/java/com/dlyagoogleplay/notes/MainActivity.kt
package com.dlyagoogleplay.notes

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.dlyagoogleplay.notes.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    //привзка макета
    private lateinit var binding: ActivityMainBinding
    //инициализируем базу данных
    private lateinit var db: NotesDatabaseHelper
    //инициализируем адаптер
    private lateinit var notesAdapter: NotesAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root) //отображаем содержимое
        //создаем базданых в этом экране
        db = NotesDatabaseHelper(this)

        //через адаптер показать в ресайкле вьюшки
        notesAdapter = NotesAdapter(db.getAllNoties(), this) //входят параметры существующего списка
        binding.notesRV.adapter = notesAdapter

        //к ресайклу прикручиваем линейный менеджер отображения
        binding.notesRV.layoutManager = LinearLayoutManager(this)

        //кнопка добавить заметку
        binding.addButton.setOnClickListener {
            val intent = Intent(this, AddNodeActivity::class.java)
            //запускается активити с заметкой
            startActivity(intent)
        }
    }
    //обновляем отображать список с новыми экзкмплярами
    override fun onResume() {
        super.onResume()
        notesAdapter.refreshData(db.getAllNoties())
        //refreshData - функция в адаптере для обновления данных
        //все новые экземпляры через адаптер  отображать в ресайкл вью
    }
}